kola/tests/sysext/zfs.go

```diff
@@ -0,0 +1,214 @@
+      [Unit]
+      ConditionFirstBoot=1
+      Before=first-boot-complete.target
+      Wants=first-boot-complete.target
+
+      [Service]
+      ExecStart=zpool create tank /dev/disk/by-id/virtio-zfs
+
+      [Install]
+      WantedBy=multi-user.target
+`)
+
+var zfsDockerUserData = conf.Butane(`
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+  - path: /etc/flatcar/enabled-sysext.conf
+    contents:
+      inline: |
+        zfs
+  - path: /etc/docker/daemon.json
+    contents:
+      inline: |
+        {
+          "storage-driver": "zfs"
+        }
+
+systemd:
+  units:
+  - name: docker.service
+    dropins:
+    - name: zfs.conf
+      contents: |
+        [Unit]
+        After=zfs.target
+
+  - name: zpool-create.service
+    enabled: true
+    contents: |
+      [Unit]
+      ConditionFirstBoot=1
+      Before=first-boot-complete.target
+      Wants=first-boot-complete.target
+
+      [Service]
+      Type=oneshot
+      ExecStart=zpool create tank /dev/disk/by-id/virtio-zfs
+      ExecStart=zfs create -o mountpoint=/var/lib/docker tank/docker
+
+      [Install]
+      WantedBy=multi-user.target
+`)
+
+var zfsNfsUserData = conf.Butane(`
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+  - path: /etc/flatcar/enabled-sysext.conf
+    contents:
+      inline: |
+        zfs
+systemd:
+  units:
+  - name: nfs-server.service
+    enabled: true
+  - name: zpool-create.service
+    enabled: true
+    contents: |
+      [Unit]
+      ConditionFirstBoot=1
+      Before=first-boot-complete.target
+      Wants=first-boot-complete.target
+
+      [Service]
+      Type=oneshot
+      ExecStart=zpool create tank /dev/disk/by-id/virtio-zfs
+      ExecStart=zfs create -o sharenfs=on tank/nfs
+
+      [Install]
+      WantedBy=multi-user.target
+`)
+
+var nfsClientUserData = conf.Butane(`
+variant: flatcar
+version: 1.0.0
+systemd:
+  units:
+  - name: nfs-client.target
+    enabled: true
+`)
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "sysext.zfs.reboot",
+		Run:         checkSysextZfs,
+		ClusterSize: 0,
+		Distros:     []string{"cl"},
+		// This test is normally not related to the cloud environment
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3902},
+	})
+
+	register.Register(&register.Test{
+		Name:        "sysext.zfs.docker",
+		Run:         checkZfsDocker,
+		ClusterSize: 0,
+		Distros:     []string{"cl"},
+		// This test is normally not related to the cloud environment
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3902},
+	})
+
+	register.Register(&register.Test{
+		Name:        "sysext.zfs.nfs",
+		Run:         checkZfsNfs,
+		ClusterSize: 0,
+		Distros:     []string{"cl"},
+		// This test is normally not related to the cloud environment
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3902},
+	})
+}
+
+func createZfsMachine(c cluster.TestCluster, userdata *conf.UserData) platform.Machine {
```

Comment:
Either here or in each top function I would add a `c.Skip` call to skip these tests on GitHub Action runs because they won't be able to download the extension. We can identify this condition by `reply.StatusCode != 200` for `reply, err := client.Head(fmt.Sprintf("https://bincache.flatcar-linux.net/images/%s/%s/version.txt", arch, version))`

Reply:
Ah, and we have to skips this for releases tests because the release can't download from the release server.
I guess it would be better to move the skipping to a temporary extra machine to first run `keySum := string(c.MustSSH(m, `md5sum /usr/share/update_engine/update-payload-key.pub.pem | cut -d " " -f 1`))` and skip `if keySum != "7192addf4a7f890c0057d21653eff2ea"`

Code extracts are from the end of `/kola/testsupdate/update.go`

Reply:
Thinking about a solution: Maybe we could inject a flag file (or kernel parameter) to the image which would tell the initrd to bind-mount the dev key instead of the prod update key, so that it would then be able to download from bincache and verify that payload.

Reply:
> I guess it would be better to move the skipping to a temporary extra machine to first run

To reduce the impact of booting a helper machine we could organize the tests under one main function

Reply:
Oh this is getting messy.

I had added a nice clean skip func implementation for GHA: https://github.com/flatcar/mantle/pull/504/commits/7d90c84c4d97a9064532950c00c3ec1cfcf5b9ee. 
But i didn't think that release tests wouldn't work.

I'll probably need to put this on hold.